test/unit/linear_algebra/test_tridiagonal_solver.cpp

```diff
@@ -52,20 +58,76 @@ TEST(TridiagSolveTest, SinglePrecision)
 /// sizes to check consistency
 TEST(TridiagSolveTest, DoublePrecision)
 {
-  std::size_t sizes[5] = { 5, 10, 1000, 100000, 10000000 };
   double error = 0;
-  for (std::size_t i = 0; i < 5; i++)
+  for (std::size_t j = 0; j < number_of_runs; j++)
   {
-    vecd x(sizes[i]);
-    vecd b(sizes[i]);
-    trid_matd A(sizes[i]);
+    std::vector<double> x(size);
+    std::vector<double> b(size);
+    tuvx::TridiagonalMatrix<double> A(size);
+
+    tuvx::FillRandom<double>(A, random_number_seed, diagonally_dominant);
+    tuvx::FillRandom<double>(x, random_number_seed);
+    b = tuvx::Dot<double>(A, x);
+    tuvx::Solve<double>(A, b);
+
+    error += tuvx::ComputeError<double>(x, b);
+  }
+  error /= number_of_runs;
+  EXPECT_LE(error, tol_dp);
+}
+
+/// @test LAPACKE Tridiagonal Solver Test for single Precision Floats.
+/// @brief Generate random tridiagonal matrix $A$ and vector $x$,
+/// compute $b=A \cdot x$, and check if solution is reconstructed
+/// accurately using L2 norm (single precision). Check for different
+/// sizes to check consistency
+TEST(LapackeTest, SinglePrecision)
+{
+  float error;
+
+  for (std::size_t j = 0; j < number_of_runs; j++)
+  {
+    std::vector<float> x(size);
+    std::vector<float> b(size);
+    tuvx::TridiagonalMatrix<float> A(size);
+
+    tuvx::FillRandom<float>(A, random_number_seed, diagonally_dominant);
+    tuvx::FillRandom<float>(x, random_number_seed);
+    b = tuvx::Dot<float>(A, x);
+
+    LAPACKE_sgtsv(
+        LAPACK_ROW_MAJOR, size, 1, A.lower_diagonal_.data(), A.main_diagonal_.data(), A.upper_diagonal_.data(), b.data(), 1);
+
+    // to be written to a file
+    error += tuvx::ComputeError<float>(x, b);
```

Comment:
Is this value being written to a file in our tests currently?

Reply:
No. I have removed this comment in my latest commit